core/chaincode/handler.go

```diff
@@ -71,6 +71,8 @@ type nextStateInfo struct {
 // Handler responsbile for management of Peer's side of chaincode stream
 type Handler struct {
 	sync.RWMutex
+	//peer to shim grpc serializer. User only in serialSend
+	serialLock  sync.RWMutex
```

Comment:
@corecode  No particular reason really. Makes sense to change it to mutex. Will do that. Thanks!